front/lib/resources/data_source_view_resource.ts

```diff
@@ -422,6 +413,32 @@ export class DataSourceViewResource extends ResourceWithVault<DataSourceViewMode
     }
   }
 
+  async destroy(auth: Authenticator, transaction?: Transaction) {
```

Comment:
Same here I would suggest not adding a new verb (very hard to quickly grasp delete vs destroy so likely error prone).

Reply:
Also we don't really need the delete/destroy for dataSourceView

So I would do

delete(hardDelete: boolean) 

If called with false error since it makes no sense for dataSourceView

Reply:
Actually it does make sense here as well, we don't want to delete all related agent configurations on a single API call. It will be slow as well.